src/modules/blogs/blogs.service.ts

```diff
@@ -0,0 +1,79 @@
+import { BadRequestException, Injectable, InternalServerErrorException, NotFoundException } from '@nestjs/common';
+import { CreateBlogDto } from './dto/create-blog.dto';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Blog } from './entities/blog.entity';
+import { Repository } from 'typeorm';
+import { User } from '../user/entities/user.entity';
+
+@Injectable()
+export class BlogsService {
+  constructor(
+    @InjectRepository(Blog)
+    private blogRepository: Repository<Blog>,
+    @InjectRepository(User)
+    private userRepository: Repository<User>
+  ) {}
+
+  async createBlog(createBlogDto: CreateBlogDto, user: User): Promise<Blog> {
+    const fullUser = await this.userRepository.findOne({
+      where: { id: user.id },
+      select: ['first_name', 'last_name'],
+    });
+
+    if (!fullUser) {
+      throw new Error('User not found');
```

Comment:
please dont throw generic error,  use Custom http exception and return appropriate message and status code, follow the guidlines on the chanel